ruby-service.sql

```diff
@@ -0,0 +1,79 @@
+-- Drop previous versions of the tables if they they exist, in reverse order of foreign keys.
+
+DROP TABLE IF EXISTS Events;
+DROP TABLE IF EXISTS Trailer;
+DROP TABLE IF EXISTS Item;
+Drop TABLE IF EXISTS TrailerUser;
+DROP TABLE IF EXISTS TrailerItem;
+DROP TABLE IF EXISTS Users;
+
+-- Create the schema.
+CREATE TABLE Users (
+  username varchar(25) PRIMARY KEY NOT NULL,
+  password varchar(50) NOT NULL,
+	type varchar(10) NOT NULL
+	);
+
+CREATE TABLE Events (
+	ID integer PRIMARY KEY, 
+	name varchar(50),
+  time timestamp,
+  description text
+	);
+
+CREATE TABLE Trailer (
+  ID integer PRIMARY KEY, 
+	name varchar(25)
+	);
+
+CREATE TABLE Item (
+  ID integer PRIMARY KEY, 
+  name varchar(25),
+  quantity integer,
+  notificationlevel integer,
+  increment integer
+  );
+
+CREATE TABLE TrailerUser (
+  TID integer REFERENCES Trailer(ID),
+  UID varchar(25) REFERENCES Users(username)
+  );
+
+CREATE TABLE TrailerItem (
+  TID integer REFERENCES Trailer(ID),
+  IID integer REFERENCES Item(ID)
+  );
+
+-- Allow users to select data from the tables.
+GRANT SELECT ON Users TO PUBLIC;
+GRANT SELECT ON Events TO PUBLIC;
+GRANT SELECT ON Trailer TO PUBLIC;
+GRANT SELECT ON Item TO PUBLIC;
+GRANT SELECT ON TrailerUser TO PUBLIC;
+GRANT SELECT ON TrailerItem TO PUBLIC;
+
+INSERT INTO Trailer VALUES (1, 'Trailer 1');
```

Comment:
Would it make sense to separate out the inserting into a different file from the table creation? That way it would be easier to recreate the tables without the test data

Reply:
I don't have a strong opinion on it. We want an easy way to test it without data present (defaults) and easily add test data as long as we're testing to make sure that the app functions properly. I'm not sure what the best way to do that. Maybe database commands to load or delete test data?